File: golang-backend/internal/handlers/attachmenthandler/dto.go
package attachmenthandler

import "github.com/aportela/doneo/internal/handlers/userhandler"

type AttachmentResponse struct {
	ID          string                       `json:"id"`
	CreatedBy   userhandler.UserBaseResponse `json:"createdBy"`
	CreatedAt   int64                        `json:"createdAt"`
	Filename    string                       `json:"name"`
	ContentType string                       `json:"contentType"`
	Size        uint32                       `json:"size"`
}

type searchResponse struct {
	ProjectAttachments []AttachmentResponse `json:"attachments"`
}

File: vite-vue-ts-frontend/src/modules/attachments/types/dto.ts
import type { UserBaseResponse } from "../../users/types/dto";

export type AttachmentResponse = {
  id: string;
  createdBy: UserBaseResponse;
  createdAt: number;
  name: string;
  contentType: string;
  size: number;
};

export type SearchResponse = {
  attachments: AttachmentResponse[];
};
